test/validation/api/ipsec/ipsec.c

```diff
@@ -1005,11 +1032,100 @@ void ipsec_check_out_in_one(const ipsec_test_part *part,
 				       pkt_in.len,
 				       pkt_in.data);
 		part_in.pkt_in = &pkt_in;
-		ipsec_check_in_one(&part_in, sa_in);
+		if (sa_in != ODP_IPSEC_SA_INVALID)
+			ipsec_check_in_one(&part_in, sa_in);
+		else
+			ipsec_send_pkts(&part_in, 1);
 		odp_packet_free(pkto[i]);
 	}
 }
 
+/**
+ * Checks the received packets for reassembly and errors.
+ */
+void ipsec_reass_out_check_in_frags(const ipsec_test_part part[],
+				    odp_ipsec_sa_t in_sa,
+				    int num_part)
+{
+	odp_queue_t queue = ODP_QUEUE_INVALID;
+	int i, j;
+
+	for (i = 0; i < num_part; i++) {
+		int num_out = part[i].num_pkt_out;
+		odp_packet_t pkto[num_out];
+
+		for (j = 0; j < num_out;) {
+			odp_packet_reass_partial_state_t reass_state;
+			odp_packet_reass_status_t reass_status;
+			odp_event_subtype_t subtype;
+			odp_packet_t pkt;
+			odp_event_t ev;
+
+			ev = odp_schedule(&queue, ODP_SCHED_NO_WAIT);
```

Comment:
The completion queue is a plain queue so one cannot dequeue from it using odp_schedule() but must use odp_queue_deq().